File: sky/src/main/java/com/skytree/epubtest/MainActivity.java
package com.skytree.epubtest;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.os.Environment;
import android.os.Handler;
import android.support.annotation.Nullable;

import com.skytree.epub.BookInformation;
import com.skytree.epub.KeyListener;

import java.io.File;

import static com.skytree.epub.Setting.debug;

/**
 * Created by keith on 2017/10/25.
 */

@Deprecated
public class MainActivity extends Activity {
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        SkyApplicationHolder app = new SkyApplicationHolder();
        Context context = this;
        app.init(context);

        String path=Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).getAbsolutePath()
                + File.separator+"books"+File.separator+"2017.epub";
        if(!new File(path).exists()){
            LocalServiceTool.debug("!new File(path).exists()");
            return;
        }

        BookInformation bi = LocalServiceTool.installBook(path);
        LocalServiceTool.openBookViewer(context,bi,true);


    }






}

File: app/src/main/java/com/tcf/epub/MainActivity.kt
package com.tcf.epub

import android.os.Bundle
import android.os.Environment
import android.support.v7.app.AppCompatActivity
import com.skytree.epub.BookInformation
import com.skytree.epubtest.LocalServiceTool
import com.skytree.epubtest.SkyApplicationHolder
import java.io.File

/**
 * Created by keith on 2017/10/25.
 */
class MainActivity:AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)



        val app = SkyApplicationHolder()
        val context = this
        app.init(context)

        val path = (Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS).absolutePath
                + File.separator + "books" + File.separator + "2017.epub")
        if (!File(path).exists()) {
            LocalServiceTool.debug("!new File(path).exists()")
            return
        }

        val bi = LocalServiceTool.installBook(path) as BookInformation

        LocalServiceTool.openBookViewer(context, bi, true)
    }
}
